src/linux/mod.rs

```diff
@@ -88,7 +88,7 @@ impl<'a> Segment<'a> {
         &self,
         shlib: &SharedLibrary<'a>,
     ) -> impl Iterator<Item = (libc::Elf32_Word, &'a [u8], &'a [u8])> {
-        assert!(self.is_note());
+        let is_note = self.is_note();
```

Comment:
I guess this doesn't hurt, but I think we only ever call this function for note segments. The other options would be to change to debug_assert or delete it.

Reply:
I prefer to delete it.